net_transport.go

```diff
@@ -626,6 +635,11 @@ func (n *NetworkTransport) handleConn(connCtx context.Context, conn net.Conn) {
 			if err != io.EOF {
 				n.logger.Error("failed to decode incoming command", "error", err)
 			}
+			if r.Buffered() > 0 {
+				snippet := [100]byte{}
+				cnt, _ := r.Read(snippet[:])
+				n.logger.Error("remaining read buffer", "sz", cnt, "data", string(snippet[:cnt]))
```

Comment:
I don't think it's safe to log unknown payloads as they may contain secrets (although it seems _exceeding_ unlikely given the first 100 bytes are likely command headers).

I think if you want more context for debugging we should add it to the errors returned by handleCommand so context can be added on a case-by-case basis.

Reply:
I mean fair, but I have no idea what the case by case basis would look like -- my primary concern is tracking down exactly where gibberish is being injected.  I suspect it is the inappropriate buffer use addressed elsewhere in this PR, but have not been able to pin it down.

Reply:
Ah sorry I see the complication: other than the initial `ReadByte` (which could use some extra context but is unlikely to be the problem), all other errors would come directly from `codec.Decoder` and largely outside of our control to annotate.

The one exception there would be wrapping Decode error return values with the `rpcType` which might be a helpful bit of context.

To add more debugging beyond that I think we need to make it configurable (and `off` by default) so that production environments can avoid leaking secrets to logs.

Reply:
(There's also some opportunity to annotate the response encoding errors with more context, but I suspect that's unrelated to the immediate problem you're trying to debug: https://github.com/hashicorp/raft/blob/f207c248e7ab7b8f6c3d7d648f11737b53e22352/net_transport.go#L755-L761)